### PR TITLE
Add targets to build the new C++ part

### PR DESCRIPTION
**Pull request by rlnx**
**Summary:**
- Add oneAPI example for PCA;
- Propagate oneAPI C++ examples to release directory;
- Add support for one-CPU build for new part;
- Split the make targets:
   - `daal` builds the old part (including java);
   - `daal_c` builds the old C++ part;
   - `onedal` builds the old part and the new one (`daal` + `oneapi`);
   - `onedal_c` build the old C++ part and the new one (`daal_c` + `oneapi`);

:penguin: **Tested only on Linux** :penguin:

---------------------------------------------------
:hammer: The recommended workflow for developers:
1. To save your time, edit `makefile.lst` to keep the algorithms needed for building the new part.
   Currently it's only PCA:
   ```make
   CORE.ALGORITHMS.FULL := pca
   ```

   In the middle of `makefile.lst` you may find out the following:
   ```make
   # Dependencies between oneAPI and core (CPU-only) algorithms
   ONEAPI.ALGOS.pca := CORE.pca

   # List of algorithms in oneAPI part
   ONEAPI.ALGOS := \
       pca
   ```

   It controls the list of algorithms to be included to the _new interfaces_. Intent of the `ONEAPI.ALGOS.pca` variable is to configure dependencies between old algorithm and new. It's not used now, but it shall help to avoid manual editing of the `CORE.ALGORITHMS.FULL` in the future . The list of required algorithms for the old part will be configured using `ONEAPI.ALGOS`. **If you want to help to implement this feature, you are welcome!** 😄 

2. New part can build at least with GCC 7.4, so
   ```sh
   WANT_GCC_VERSION=7.4.0 source lnx32e.sh
   make -j24 onedal_c REQCPU=avx2/avx512
   ```

   GCC 7.4 is a minimal supported version, but I'd recommend to use GCC 9.3 for better productivity as it provides much better error reporting especially for templates:
   ```sh
   WANT_GCC_VERSION=9.3.0
   ``` 

----------------------------------------------------
:wrench: The build in CI can be organized as follows:
```sh
WANT_GCC_VERSION=4.8.5 source lnx32e.sh
make daal  # builds only the old part

WANT_GCC_VERSION=7.4.0 source lnx32e.sh
make onedal # build only the new part as the old one is already built
```

### CODE REVIEW

File: cpp/oneapi/dal/backend/dispatcher_cpu.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/backend/dispatcher_cpu.hpp:2]: 2014?

Comment (rlnx) [reply cpp/oneapi/dal/backend/dispatcher_cpu.hpp:2]: This is a problem for whole new part, headers are copy-pasted from the old one. It should be fixed in separate PR. I'll create it as there are many things need to be cleaned up.

**Comment by rlnx** [comment]
@mchernov-intel, @michael-smirnov could you please test the changes locally? It should unblock you for adding your portion of functionality. The new part is not enabled in CI yet, @napetrov will experiment with it.

**Comment by mchernov-intel** [comment]
@rlnx in couple of hours

**Comment by napetrov** [comment]
/intelci: restart
set infra_branch=dev/napetrov_enable_onedal

**Comment by napetrov** [comment]
Let's also switch public ci to onedal make target - https://github.com/rlnx/daal/blob/enable-onedal/.ci/scripts/build.sh#L77

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/napetrov_enable_onedal

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/napetrov_enable_onedal

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/napetrov_enable_onedal

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/napetrov_enable_onedal

**Comment by rlnx** [comment]
/intelci: restart
set infra_branch=dev/napetrov_enable_onedal

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/napetrov_enable_onedal

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/napetrov_enable_onedal

**Pull request merged**